crypto/rand/randfile.c

```diff
@@ -124,9 +138,9 @@ int RAND_load_file(const char *file, long bytes)
 
     for ( ; ; ) {
         if (bytes > 0)
-            n = (bytes < RAND_FILE_SIZE) ? (int)bytes : RAND_FILE_SIZE;
+            n = (bytes <= RAND_LOAD_BUF_SIZE) ? (int)bytes : RAND_BUF_SIZE;
```

Comment:
did you mean `? (int)bytes : RAND_BUF_SIZE;` ?

Reply:
I dont' understand your question, ist the quoted part incomplete?

Just in case you are wondering why there are two different constants (`RAND_LOAD_BUF_SIZE` and `RAND_BUF_SIZE`), please read the comment at https://github.com/openssl/openssl/pull/7456/files/b75b060a4524288478ef5caa2c8192a4af8c65ef#diff-1014acebaa2c13d44ca196b9a433ef2eR78 for an explanation.

Reply:
Oh, and what happens if 1 <= bytes <= RAND_DRBG_STRENGTH-1 ?

Reply:
This case is treated in b75b060a4524288478ef5caa2c8192a4af8c65ef, see the commit message.

Short version: if `bytes < RAND_DRBG_STRENGTH` then I will add the random input as additional_data only, not as entropy input. Reseeding will work always if an os entropy source is available.  If `--with-rand-seed=none` than things get trickier but I try to avoid failure as much as possible. But there are theoretical cases where the reseed can fall. But there's nothing one can do about it if there's too little entropy.

Reply:
Oh, and what happens if the read below returns less bytes than requested ?

Reply:
Yes, now I think that is fine, I was just curious.  Thanks for you explanations.